status-page.py

```diff
@@ -1,72 +1,133 @@
+
+    def get_all_open_incidents(self):
         response = requests.get(f'{self.base_url}/incidents', self.header)
-        print(response)
         for incident in response.json()['data']:
             status = incident['status']
-            if status == 1:
-                self.incidents['investigating'].append(incident)
-            if status == 2:
-                self.incidents['identified'].append(incident)
-            if status == 3:
-                self.incidents['watching'].append(incident)
-            if status == 4:
-                self.incidents['fixed'].append(incident)
+            if status != 4:
+                self.open_incidents.append(incident)
+
+    def update_incident_status(self):
+        # map api version and name to Cachet component IDs
+        component_map = {
+            'v1/academic-disciplines': 3,
+            'v1/advisors': 4,
+            'v1/xeapps': 5,
+            'v1/beaverbus': 6,
+            'v1/directory': 7,
+            'v2/directory': 8,
+            'v1/finance': 9,
+            'v1/hr': 10,
+            'v1/identify': 11,
+            'v1/locations': 12,
+            'v1/oauth2': 13,
+            'v1/onbase': 14,
+            'v1/persons': 15,
+            'v1/staff-fee-privilege': 16,
+            'v1/students': 17,
+            'v1/terms': 18,
+            'v1/textbooks': 19
+        }
+
+        # then, if there are any test failures, open an incident if there is none open already
+        for failed_test in self.log_json['failed_tests']:
+            already_reported = False
+            parsed_url = urlparse(failed_test['base_url'])
+            split_path = parsed_url.path.split('/')
+            api_version = split_path[1]
+            api_name = split_path[2]
+            version_and_name = f'{api_version}/{api_name}'
 
-    # Should we update incident status to resolved if test succeeds on API
-    # with open issue?
-    def updateStatuses(self):
-        print("Updating Statuses")
+            if version_and_name in component_map:
+                component_id = component_map[version_and_name]
+                for incident in self.open_incidents:
+                    if component_id == incident['component_id']:
+                        already_reported = True
+                if not already_reported:
+                    body = {
+                        'name': failed_test['base_url'],
+                        'message': f'Jenkins has reported an issue with {api_name} api {api_version}.\
+                                     Failed test url: {failed_test["base_url"]}',
+                        'status': 1,
+                        'visible': 1,
+                        'component_id': component_id,
+                        'component_status': 2,
+
+                    }
+                    print('Posting New Incident:')
+                    response = requests.post(f'{self.base_url}/incidents', headers=self.header, data=body)
+                    print(response)
```

Comment:
These print statement should be able to be combined to just one print statement with multi-line strings.